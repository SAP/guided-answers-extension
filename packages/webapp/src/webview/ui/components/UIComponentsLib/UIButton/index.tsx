import { IButton } from '@fluentui/react';

export { IButton as UIButtonInterface };
export * from './UIDefaultButton';
export * from './UIIconButton';
export * from './UIActionButton';
