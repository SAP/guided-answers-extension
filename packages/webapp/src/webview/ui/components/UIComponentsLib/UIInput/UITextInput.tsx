import React from 'react';

import type { ITextFieldStyleProps, ITextFieldStyles } from '@fluentui/react';
import { TextField } from '@fluentui/react';

export { ITextField, ITextFieldProps } from '@fluentui/react';

/**
 * UITextInput component
 * based on https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield
 *
 * @export
 * @class UITextInput
 * @extends {React.Component<ITextFieldProps, {}>}
 */
export class UITextInput extends React.Component<any> {
    public constructor(props: any) {
        super(props);
    }

    render(): JSX.Element {
        const textFieldStyles = (props: ITextFieldStyleProps): Partial<ITextFieldStyles> => {
            return {
                ...{
                    root: {
                        height: 'auto',
                        fontFamily: 'var(--vscode-font-family)'
                    },
                    fieldGroup: [
                        !props.disabled &&
                            !props.multiline && {
                                backgroundColor: 'var(--vscode-input-background)',
                                border: '1px solid var(--vscode-editorWidget-border)',
                                borderRadius: 0,
                                height: 24,
                                maxHeight: 24,
                                minHeight: 24,
                                boxSizing: 'initial',
                                selectors: {
                                    '&:hover': {
                                        border: '1px solid var(--vscode-focusBorder)'
                                    }
                                }
                            },
                        props.multiline && {
                            backgroundColor: 'var(--vscode-input-background)',
                            border: '1px solid var(--vscode-editorWidget-border)',
                            borderRadius: 0,
                            boxSizing: 'initial',
                            selectors: {
                                '&:hover': {
                                    border: '1px solid var(--vscode-focusBorder)'
                                }
                            }
                        },
                        props.disabled &&
                            !props.multiline && {
                                color: 'var(--vscode-input-foreground)',
                                opacity: 0.4,
                                backgroundColor: `var(--vscode-editor-inactiveSelectionBackground)`,
                                borderColor: 'var(--vscode-editorWidget-border)',
                                border: '1px solid var(--vscode-editorWidget-border)',
                                borderRadius: 0,
                                height: 24,
                                maxHeight: 24,
                                minHeight: 24,
                                boxSizing: 'initial'
                            },
                        props.focused &&
                            !props.disabled && {
                                selectors: {
                                    ':after': {
                                        border: `1px solid var(--vscode-focusBorder)`
                                    }
                                }
                            }
                    ],
                    field: [
                        !props.multiline && {
                            backgroundColor: 'var(--vscode-input-background)',
                            color: 'var(--vscode-input-foreground)',
                            lineHeight: 'normal',
                            height: 24,
                            maxHeight: 24,
                            minHeight: 24,
                            fontSize: '13px',
                            fontWeight: 'normal',
                            boxSizing: 'border-box',
                            selectors: {
                                '::placeholder': {
                                    fontSize: 13,
                                    fontFamily: 'var(--vscode-font-family)',
                                    color: 'var(--vscode-input-placeholderForeground)'
                                }
                            }
                        },
                        props.hasIcon && {
                            selectors: {
                                '&:hover': {
                                    cursor: 'pointer'
                                }
                            }
                        },

                        props.multiline && {
                            backgroundColor: 'var(--vscode-input-background)',
                            color: 'var(--vscode-input-foreground)',
                            minHeight: '60px',
                            height: 'auto',
                            display: 'flex',
                            fontSize: '13px',
                            fontWeight: 'normal',
                            boxSizing: 'border-box',
                            selectors: {
                                '::placeholder': {
                                    fontSize: 13,
                                    fontFamily: 'var(--vscode-font-family)',
                                    color: 'var(--vscode-input-placeholderForeground)'
                                }
                            }
                        },

                        // props for input element inside field group
                        props.disabled && {
                            fontStyle: 'italic',
                            backgroundColor: 'transparent'
                        }
                    ],
                    suffix: {
                        backgroundColor: 'var(--vscode-input-background)'
                    },
                    subComponentStyles: {
                        label: {
                            root: [
                                {
                                    marginTop: 25
                                },
                                props.disabled && {
                                    opacity: '0.4'
                                },
                                props.required && {
                                    selectors: {
                                        '::after': {
                                            content: `' *'`,
                                            color: 'var(--vscode-inputValidation-errorBorder)',
                                            paddingRight: 12
                                        }
                                    }
                                }
                            ]
                        }
                    },
                    icon: [
                        {
                            bottom: 2
                        }
                    ]
                }
            };
        };

        return <TextField {...this.props} styles={textFieldStyles} />;
    }
}
