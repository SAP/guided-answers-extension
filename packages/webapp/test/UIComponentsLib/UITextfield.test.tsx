import * as React from 'react';
import * as Enzyme from 'enzyme';
import type { IStyleFunction, ITextFieldStyleProps } from '@fluentui/react';
import { TextField } from '@fluentui/react';
import { UITextInput } from '../../src/webview/ui/components/UIComponentsLib/UIInput';

describe('<UIToggle />', () => {
    let wrapper: Enzyme.ReactWrapper<any>;

    beforeEach(() => {
        wrapper = Enzyme.mount(<UITextInput />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    it('Should render a UITextInput component', () => {
        expect(wrapper.find('.ms-TextField').length).toEqual(1);
    });

    it('Styles - default', () => {
        const styles = (wrapper.find(TextField).props().styles as IStyleFunction<{}, {}>)({}) as ITextFieldStyleProps;
        expect(styles).toMatchSnapshot();
    });
});
