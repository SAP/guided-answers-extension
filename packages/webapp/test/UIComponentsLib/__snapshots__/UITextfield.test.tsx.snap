// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UIToggle /> Styles - default 1`] = `
Object {
  "field": Array [
    Object {
      "backgroundColor": "var(--vscode-input-background)",
      "boxSizing": "border-box",
      "color": "var(--vscode-input-foreground)",
      "fontSize": "13px",
      "fontWeight": "normal",
      "height": 24,
      "lineHeight": "normal",
      "maxHeight": 24,
      "minHeight": 24,
      "selectors": Object {
        "::placeholder": Object {
          "color": "var(--vscode-input-placeholderForeground)",
          "fontFamily": "var(--vscode-font-family)",
          "fontSize": 13,
        },
      },
    },
    undefined,
    undefined,
    undefined,
  ],
  "fieldGroup": Array [
    Object {
      "backgroundColor": "var(--vscode-input-background)",
      "border": "1px solid var(--vscode-editorWidget-border)",
      "borderRadius": 0,
      "boxSizing": "initial",
      "height": 24,
      "maxHeight": 24,
      "minHeight": 24,
      "selectors": Object {
        "&:hover": Object {
          "border": "1px solid var(--vscode-focusBorder)",
        },
      },
    },
    undefined,
    undefined,
    undefined,
  ],
  "icon": Array [
    Object {
      "bottom": 2,
    },
  ],
  "root": Object {
    "fontFamily": "var(--vscode-font-family)",
    "height": "auto",
  },
  "subComponentStyles": Object {
    "label": Object {
      "root": Array [
        Object {
          "marginTop": 25,
        },
        undefined,
        undefined,
      ],
    },
  },
  "suffix": Object {
    "backgroundColor": "var(--vscode-input-background)",
  },
}
`;
